src/__tests__/reactUtils-test.js

```diff
@@ -0,0 +1,232 @@
+      return result;
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    async sleep(ms: number): Promise<void> {
+      // The hook uses `useState`, which seems to make `react-test-renderer`
+      // complain if we don't use `act`.
+      return act(() => fakeSleep(ms));
+    }
+
+    cleanup() {
+      // https://reactjs.org/docs/test-renderer.html#testrendererunmount
+      this._testRenderer.unmount();
+    }
+
+    _createTestRenderer(initialValue: boolean) {
+      const TestComponent = this._TestComponent;
+      let testRenderer;
+      act(() => {
+        // https://reactjs.org/docs/test-renderer.html#testrenderercreate
+        testRenderer = create(<TestComponent value={initialValue} />);
+      });
+      return testRenderer;
+    }
+
+    _updateTestRenderer(value: boolean) {
+      const TestComponent = this._TestComponent;
+      act(() => {
+        // https://reactjs.org/docs/test-renderer.html#testrendererupdate
+        this._testRenderer.update(<TestComponent value={value} />);
+      });
+    }
+  }
+
+  const MS = 1000;
+
+  /**
+   * Simulate the input value changing over time, checking the hook's output.
+   *
+   * On each item in the `sequence`, this will:
+   * 1. Wait for a specified time
+   * 2. Read and assert the hook's output from the last render, as specified
+   * 3. Render again, with the specified input for the hook
+   */
+  // Tell ESLint to recognize `testSequence` as a helper function that runs
+  // assertions.
+  /* eslint jest/expect-expect: ["error", { "assertFunctionNames": ["expect", "testSequence"] }] */
+  const testSequence = async (args: {
+    initialValue: boolean,
+    sequence: $ReadOnlyArray<{|
+      waitBefore: number,
+      expectedOutput: boolean,
+      thenUpdateInputTo?: boolean,
+    |}>,
+  }) => {
+    const { initialValue, sequence } = args;
+    const testMachine = new TestMachine(MS, initialValue);
+
+    // Should never be true before any time has passed.
+    expect(testMachine.hookOutput()).toBeFalse();
+
+    for (let i = 0; i < sequence.length; i++) {
+      const { waitBefore, expectedOutput, thenUpdateInputTo } = sequence[i];
+      await testMachine.sleep(waitBefore);
+      expect(testMachine.hookOutput()).toBe(expectedOutput);
+      if (thenUpdateInputTo !== undefined) {
+        testMachine.updateValue(thenUpdateInputTo);
+      }
+    }
+
+    testMachine.cleanup();
+  };
+
+  const sequencesToTest = [
+    {
+      initialValue: false,
+      sequence: [{ waitBefore: MS * 2, expectedOutput: false }],
+    },
+    {
+      initialValue: false,
+      sequence: [{ waitBefore: MS / 2, expectedOutput: false }],
+    },
+    {
+      initialValue: false,
+      sequence: [
+        { waitBefore: MS / 2, expectedOutput: false, thenUpdateInputTo: true },
+        { waitBefore: MS * 2, expectedOutput: true, thenUpdateInputTo: false },
+        { waitBefore: MS / 2, expectedOutput: false },
```

Comment:
nit:
```suggestion
      sequence: [
        { waitBefore: MS / 2, expectedOutput: false, thenUpdateInputTo: true },
        { waitBefore: 2 * MS, expectedOutput: true, thenUpdateInputTo: false },
        { waitBefore: MS / 2, expectedOutput: false },
```
and similarly the other `MS * 2` instances.

That makes the "twice" and "half" more visually distinct, making it a bit easier to scan each of these scenarios and see what it's saying. Perhaps not coincidentally, it's also more conventional in math and physics: numeric coefficient first, then the variable or dimensional quantity it's multiplying goes after that.